gitlab/table_my_event.go

```diff
@@ -0,0 +1,191 @@
+		List: &plugin.ListConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:      "created_at",
+					Require:   plugin.Optional,
+					Operators: []string{">", ">=", "=", "<", "<="},
+				},
+				{
+					Name:      "target_type",
+					Require:   plugin.Optional,
+					Operators: []string{"="},
+				},
+				{
+					Name:      "action_name",
+					Require:   plugin.Optional,
+					Operators: []string{"="},
+				},
+			},
+			Hydrate: listMyEvents,
+		},
+		Columns: myEventColumns(),
+	}
+}
+
+// Hydrate Functions
+func listMyEvents(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Debug("listMyEvents", "started")
+	conn, err := connect(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("listMyEvents", "unable to establish a connection", err)
+		return nil, fmt.Errorf("unable to establish a connection: %v", err)
+	}
+
+	opt := &api.ListContributionEventsOptions{ListOptions: api.ListOptions{
+		Page:    1,
+		PerPage: 50,
+	}}
+
+	if d.Quals["created_at"] != nil {
+		for _, q := range d.Quals["created_at"].Quals {
+			givenTime := q.Value.GetTimestampValue().AsTime()
+			beforeTime := givenTime.Add(time.Duration(-1) * time.Second)
+			afterTime := givenTime.Add(time.Second * 1)
+			givenISOTime := api.ISOTime(givenTime)
+			beforeISOTime := api.ISOTime(beforeTime)
+			afterISOTime := api.ISOTime(afterTime)
+
+			switch q.Operator {
+			case ">":
+				opt.After = &afterISOTime
+			case ">=":
+				opt.After = &givenISOTime
+			case "=":
+				opt.After = &beforeISOTime
+				opt.Before = &afterISOTime
+			case "<=":
+				opt.Before = &givenISOTime
+			case "<":
+				opt.Before = &beforeISOTime
+			}
+		}
+	}
+
+	if d.Quals["target_type"] != nil {
+		targetType := api.EventTargetTypeValue(
+			strings.ToLower(d.EqualsQuals["target_type"].GetStringValue()))
+		opt.TargetType = &targetType
+	}
+
+	if d.Quals["action_name"] != nil {
+		action := api.EventTypeValue(
+			strings.ToLower(d.EqualsQuals["action_name"].GetStringValue()))
+		opt.Action = &action
+	}
+
+	for {
+		plugin.Logger(ctx).Debug("listMyEvents", "page", opt.Page, "perPage", opt.PerPage)
+		events, resp, err := conn.Events.ListCurrentUserContributionEvents(opt)
+		if err != nil {
+			plugin.Logger(ctx).Error("listMyEvents", "page", opt.Page, "error", err)
+			return nil, fmt.Errorf("unable to obtain my events\n%v", err)
+		}
+
+		for _, event := range events {
+			d.StreamListItem(ctx, event)
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+
+		opt.Page = resp.NextPage
+	}
+
+	plugin.Logger(ctx).Debug("listMyEvents", "completed successfully")
+	return nil, nil
+}
+
+// Column Function
+func myEventColumns() []*plugin.Column {
+	return []*plugin.Column{
+		{
+			Name:        "id",
+			Type:        proto.ColumnType_INT,
+			Description: "The event ID",
+		},
+		{
+			Name:        "title",
+			Type:        proto.ColumnType_STRING,
+			Description: "Possibly abandoned title of the event",
```

Comment:
Not really sure on this column description, what's the meaning of `Possibly abandoned` here?

Reply:
It always seems to be null in the api.  I've tried a bunch of queries.  The gitlab docs are silent.  I'll go look through the gitlab code to figure it out.  I really hate finding things in ruby apis, but it's on my todo list.

Reply:
OK, hunting around the issues I see this: https://gitlab.com/gitlab-org/gitlab/-/issues/197706

It devolves into a "we should use swagger for documentation" discussion, but it seems that it's not used.  Would it be better just not to include it?

Reply:
OK, this finally finished and I see this on gitlab.com (`m.project_id = 278964:` is the gitlab project) and it really seems unused.  I'll just remove it and leave a comment.

```
select event.title,count(event.*) from gitlab.gitlab_project_member as m, gitlab.gitlab_user_event as event where event.created_at > '2023-08-01' and event.author_id = m.id and m.id % 20 = 2 and ` group by event.title;
+--------+-------+
| title  | count |
+--------+-------+
| <null> | 6781  |
+--------+-------+
```

Reply:
Oh, and the `m.id % 20 = 2` bit is just to limit the api calls. There 2k members so `% 20 = 2` should limit it to 100 users.